src/components/Menu/Menu.tsx

```diff
@@ -1,6 +1,94 @@
-import React from 'react';
+import React, { useState, useRef } from 'react';
+import Banner from '../Banner/Banner';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faChevronLeft,
+  faChevronRight,
+} from '@fortawesome/free-solid-svg-icons';
+import menu from './menu.json';
 import './Menu.css';
 
-const Menu = () => <div className="menu"></div>;
+interface ICategory {
+  name: IMultiLanguageName;
+  image: string;
+  items: IItem[];
+}
+
+interface IMultiLanguageName {
+  en: string;
+  zh: string;
+  vi: string;
+}
+
+interface IItem {
+  number: string;
+  name: IMultiLanguageName;
+  price: string;
+}
+
+enum Languages {
+  ENGLISH = 'en',
+  CHINESE = 'zh',
+  VIETNAMESE = 'vi',
+}
+
+const Menu = () => {
+  const [language, setLanguage] = useState(Languages.ENGLISH);
+  const [activeCategory, setActiveCategory] = useState<ICategory>(
+    menu.categories[0]
+  );
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  const scroll = (amount: number) =>
+    sliderRef.current?.scrollTo({
+      left: sliderRef.current.scrollLeft + amount,
+      behavior: 'smooth',
+    });
+
+  return (
+    <div className="menu">
+      <Banner message={menu.delivery} tooltipMessages={menu.discounts} />
+      <div className="slider-container">
+        <div className="left" onClick={() => scroll(-500)}>
+          <FontAwesomeIcon icon={faChevronLeft} size="lg" />
+        </div>
+        <div ref={sliderRef} className="slider">
+          {menu.categories.map((category: ICategory, index) => (
+            <div
+              key={index}
```

Comment:
I would suggest using something other than `index` as the key, as it's not reliable. Maybe `category.name[language]`?

Reply:
After reading up on keys a bit I would argue that `index` as key makes sense for static lists (nothing changes unless `menu.json` changes). Thoughts?

Reply:
Yea that's fine then as `category.name[language]` could be clunky. If there were no issues with this during testing it should be fine to use